compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilCompilationException.kt

```diff
@@ -0,0 +1,96 @@
+package com.squareup.anvil.compiler.api
+
+import org.jetbrains.kotlin.codegen.CompilationException
+import org.jetbrains.kotlin.com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.com.intellij.psi.PsiNameIdentifierOwner
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.descriptors.annotations.AnnotationDescriptor
+import org.jetbrains.kotlin.ir.IrElement
+import org.jetbrains.kotlin.ir.declarations.IrClass
+import org.jetbrains.kotlin.ir.symbols.IrSymbol
+import org.jetbrains.kotlin.ir.util.render
+import org.jetbrains.kotlin.js.resolve.diagnostics.findPsi
+import org.jetbrains.kotlin.resolve.source.KotlinSourceElement
+import org.jetbrains.kotlin.resolve.source.getPsi
+import org.jetbrains.kotlin.util.getExceptionMessage
+
+/**
+ * `AnvilCompilationException` is thrown when Anvil specific can't be processed or an error
+ * occurs while generating code.
+ */
+public class AnvilCompilationException(
+  message: String,
+  cause: Throwable? = null,
+  element: PsiElement? = null
+) : CompilationException(message, cause, element) {
+  public companion object {
+    public operator fun invoke(
+      annotationDescriptor: AnnotationDescriptor,
+      message: String,
+      cause: Throwable? = null
+    ): AnvilCompilationException {
+      return AnvilCompilationException(
+        message = message,
+        cause = cause,
+        element = annotationDescriptor.identifier
+      )
+    }
+
+    public operator fun invoke(
+      classDescriptor: ClassDescriptor,
+      message: String,
+      cause: Throwable? = null
+    ): AnvilCompilationException {
+      return AnvilCompilationException(
+        message = message,
+        cause = cause,
+        element = classDescriptor.identifier
+      )
+    }
+
+    public operator fun invoke(
+      message: String,
```

Comment:
The ordering isn't standardized across these invocations. Could you align these? 

message
cause
object

Reply:
Oh, good catch!